### PR TITLE
chore: Separate internal and external docs

### PR DESCRIPTION
**Pull request by binarylogic**
This is a proposal to move the website to a new root-level `website` directory, reserving the root-level `docs` folder for actual project docs. This is helpful as we expand our community documentation; we are planning to add a few more docs as part of introducing more policies. Containing them to a single directory makes them easier to find and understand. I'm curious what others think.

You can see the proposed tree [here](https://github.com/timberio/vector/tree/move-website).

- [x] Add `docs/README.md`
- [ ] Fix website build errors
- [x] Further audit the repository for `docs` -> `website` path changes

### CODE REVIEW

**Review by jszwedko — COMMENTED** [review]
Splitting up the website and cue definitions makes sense to me. I am used to finding `SECURITY.md`, `CONTRIBUTING.md`, etc. at the top-level though. What would you think about leaving those ones there?

**Comment by binarylogic** [comment]
I'm not opposed to it, Github [supports the `docs` directory as an official directory for these files](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors). Are you saying only these 2 files would be at the root, the others would be in `docs`? For additional context, I would like to add the following docs:

* `USER_EXPERIENCE_DESIGN.md` (split out of `CONTRIBUTING.md` and expanded significantly)
* `LEGAL.md` (split out of `CONTRIBUTING.md`)
* `DEVELOPMENT.md` (split out of `CONTRIBUTING.md`)

I know this is a lot, but it's all important. We can reference it in reviews, etc.

**Comment by lucperkins** [comment]
No quarrel from me. FYI @binarylogic changing the `build.base` parameter in the Netlify config should get the site building again.

**Comment by binarylogic** [comment]
Sure, happy to move the `cue` directory into the `website` directory, but I would like a folder for our project docs, and Github only supports the `docs` folder :/.

**Comment by lucperkins** [comment]
@binarylogic I guess I'm not quite sure what makes the CUE sources "project docs" but the Markdown docs not. To me they're a single, interlocked unit.

**Comment by binarylogic** [comment]
I completely agree. I'm trying to make the distinction between external and internal docs. I'm proposing that the `docs` folder be used for internal docs and the `website` folder be used for external docs. Github only supports the `docs` folder for special internal docs ([example](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors)).

**Comment by lucperkins** [comment]
@binarylogic Ah, I see. Sounds like keeping the `cue` directory under `docs` was an oversight, then.

Two other places that come to mind where we'd need to do the `s/docs/website` switcheroo:

* `scripts/*.rb`
* `website/scripts/cue.sh`

**Comment by jszwedko** [comment]
> I'm not opposed to it, Github [supports the `docs` directory as an official directory for these files](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors). Are you saying only these 2 files would be at the root, the others would be in `docs`? For additional context, I would like to add the following docs:

Yeah, that's a good question. I guess I'd lean towards just putting all of those MD files at the top-level as that is the most common location, in my experience. My second choice would be to put them in `docs/` but update README.md to point people there.

**Comment by binarylogic** [comment]
@jszwedko @lucperkins could I get another review? This is in a good place now. You can review the commits for a break down of the changes I made; I'm following this structure:

1. The `/docs` directory is for internal contributor docs.
2. The `/` root directory contains Vector project policies (security, privacy, code of conduct, etc)
3. The `/website` directory contains external docs.

And within the `/docs` directory I broke out `CONTRIBUTING.md` to `DEVELOPING.md` and `DOCUMENTING.md`, keeping `CONTRIBUTING.md` focus on the workflow.

What do you think?

**Comment by lucperkins** [comment]
@binarylogic I don't have any strong feelings here, so if you think this is a win, go for it.

FYI you'll need to update the `build.base` param in `netlify.toml` to get the site building again.

**Comment by binarylogic** [comment]
Thanks. @lucperkins I think it is a win since we'll be adding more policy documents shortly. It's a bit hard to follow otherwise.

**Review by spencergilbert — APPROVED** [review]
I also like the idea here

**Comment by nurlybekovnt** [comment]
@binarylogic @jszwedko @spencergilbert @lucperkins Hi! How do you draw the architecture diagrams? Here in the repository I see PNG and SVG files. What tools do you use? Where do you store the sources of these drawings?

**Comment by jszwedko** [comment]
> @binarylogic @jszwedko @spencergilbert @lucperkins Hi! How do you draw the architecture diagrams? Here in the repository I see PNG and SVG files. What tools do you use? Where do you store the sources of these drawings?

Unfortunately I'm not sure how the diagrams were created. Here they were just moved around.

**Comment by pront** [comment]
> > @binarylogic @jszwedko @spencergilbert @lucperkins Hi! How do you draw the architecture diagrams? Here in the repository I see PNG and SVG files. What tools do you use? Where do you store the sources of these drawings?
> 
> Unfortunately I'm not sure how the diagrams were created. Here they were just moved around.

FWIW I use LucidChart or Dia.
